src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/VSMSBuildNuGetProjectSystem.cs

```diff
@@ -899,7 +901,14 @@ public virtual void RegisterProcessedFiles(IEnumerable<string> files)
 
         public virtual void EndProcessing()
         {
-            ProjectBuildSystem?.EndBatchEdit();
+            if (ProjectBuildSystem != null)
+            {
+                NuGetUIThreadHelper.JoinableTaskFactory.Run(async delegate
+                {
+                    await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                    ProjectBuildSystem.EndBatchEdit();
```

Comment:
Same here: consider double-checking the value against null once you reach the UI thread to reduce risk of this escrow change.